src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const COPY_OF_STATE = { ...state };
+  const ARRAY_OF_STATES = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(COPY_OF_STATE, action.extraData);
+        break;
+
+      case 'removeProperties':
+        const removeValue = action.keysToRemove;
+
+        for (const key of Object.keys(COPY_OF_STATE)) {
+          if (removeValue.includes(key)) {
+            delete COPY_OF_STATE[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const value of Object.keys(COPY_OF_STATE)) {
+          delete COPY_OF_STATE[value];
+        }
+        break;
+    }
```

Comment:
[Checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) p. 3 - switch/case should always have default case for error handling.